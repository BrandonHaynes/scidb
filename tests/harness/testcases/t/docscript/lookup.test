
--test
--start-query-logging

--set-format dense
store(build(<val1:double>[i=0:3,4,0],1),indices1)  
store(build(<val1:double>[i=0:3,4,0],i),indices2)  
--set-format dcsv
store(join(indices1,indices2),pattern_array)  
--set-format dense
store(build(<val:double>[i=0:3,4,0,j=0:3,4,0],i*4+j+100),source_array)  
lookup(pattern_array,source_array)  

--stop-query-logging


--cleanup

remove(indices1)
remove(indices2)
remove(pattern_array)
remove(source_array)
