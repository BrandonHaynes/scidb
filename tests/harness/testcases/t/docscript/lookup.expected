SCIDB QUERY : <store(build(<val1:double>[i=0:3,4,0],1),indices1)>
[(1),(1),(1),(1)]

SCIDB QUERY : <store(build(<val1:double>[i=0:3,4,0],i),indices2)>
[(0),(1),(2),(3)]

SCIDB QUERY : <store(join(indices1,indices2),pattern_array)>
{i} val1,val1_2
{0} 1,0
{1} 1,1
{2} 1,2
{3} 1,3

SCIDB QUERY : <store(build(<val:double>[i=0:3,4,0,j=0:3,4,0],i*4+j+100),source_array)>
[[(100),(101),(102),(103)],[(104),(105),(106),(107)],[(108),(109),(110),(111)],[(112),(113),(114),(115)]]

SCIDB QUERY : <lookup(pattern_array,source_array)>
[(104),(105),(106),(107)]

Query was executed successfully

Query was executed successfully

Query was executed successfully

Query was executed successfully

